contribs/application/src/test/java/org/matsim/application/options/ShpOptionsTest.java

```diff
@@ -61,4 +63,35 @@ public void all() {
 				.hasSize(Set.copyOf(ft).size());
 
 	}
+
+	@Test
+	public void testGetGeometry() {
+
+		Path input = Path.of(utils.getClassInputDirectory()
+						.replace("ShpOptionsTest", "CreateLandUseShpTest")
+						.replace("options", "prepare"))
+				.resolve("andorra-latest-free.shp.zip");
+
+		Assume.assumeTrue(Files.exists(input));
+
+		ShpOptions shp = new ShpOptions(input, null, null);
+		Geometry geometry1 = shp.getGeometry() ;
+		Geometry geometry2 = null ;
+		Geometry geometry3 = null ;
+
+		List<SimpleFeature> features = shp.readFeatures();
+
+		geometry3 = (Geometry) features.get(0).getDefaultGeometry();
+
```

Comment:
The way you have structured this for loop was also hard to parse for me. I would rather initialize a geometry before the loop and then have no if/else statement in the loop
```
		var geometry = new GeometryFactory().createEmpty(2);
		
		for (var feature : features) {
			var geometryToJoin = (Geometry) feature.getDefaultGeometry();
			geometry = geometry.union(geometryToJoin);
		}
```

Reply:
re-structured the for loop, thanks for the advice